internal/storage/gcs/bucket.go

```diff
@@ -149,4 +149,10 @@ type Bucket interface {
 
 	// Atomically rename folder for Hierarchical bucket.
 	RenameFolder(ctx context.Context, folderName string, destinationFolderId string) (*control.RenameFolderOperation, error)
+
+	// List the folders in the Hierarchical bucket that meet the criteria defined by the
+	// request, returning a result folders that contains the results and
+	// potentially a cursor for retrieving the next portion of the larger set of
+	// results.
+	ListFolders(ctx context.Context, req *controlpb.ListFoldersRequest) (*ListingFolders, error)
```

Comment:
Can we create *controlpb.ListFoldersRequest inside bucket handle and pass only requiredvalues which are needed to make the request, only bucket handle should not know about controlpb package.. that way we could decouple external dependency code from most of our classes. For eg: if in future suppose ListFoldersRequest moves to another package or some other change comes in sdk for this strcut all our classes will have to change, if we limit it to one class and have our won structs be used in method signatures, that way we can decouple the sdk code.
its on similar lines why we have created our own folder struct., adding you @vadlakondaswetha also in case you have other thoughts around this..

Reply:
Agree with Ankita. We should not add dependencies of SDK anywhere except the bucket_handle.go code.

Reply:
Done